api/src/main/java/com/epam/pipeline/manager/quota/RunLimitsService.java

```diff
@@ -0,0 +1,176 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.manager.quota;
+
+import com.epam.pipeline.common.MessageConstants;
+import com.epam.pipeline.common.MessageHelper;
+import com.epam.pipeline.controller.vo.PipelineRunFilterVO;
+import com.epam.pipeline.entity.contextual.ContextualPreference;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceExternalResource;
+import com.epam.pipeline.entity.contextual.ContextualPreferenceLevel;
+import com.epam.pipeline.entity.pipeline.TaskStatus;
+import com.epam.pipeline.entity.user.ExtendedRole;
+import com.epam.pipeline.entity.user.PipelineUser;
+import com.epam.pipeline.entity.user.Role;
+import com.epam.pipeline.exception.quota.LaunchQuotaExceededException;
+import com.epam.pipeline.manager.contextual.ContextualPreferenceManager;
+import com.epam.pipeline.manager.pipeline.PipelineRunManager;
+import com.epam.pipeline.manager.preference.SystemPreferences;
+import com.epam.pipeline.manager.security.AuthManager;
+import com.epam.pipeline.manager.user.RoleManager;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RunLimitsService {
+
+    private static final List<TaskStatus> ACTIVE_RUN_STATUSES = new ArrayList<>(Arrays.asList(TaskStatus.RESUMING,
+                                                                                              TaskStatus.RUNNING));
+    private final PipelineRunManager runManager;
+    private final RoleManager roleManager;
+    private final ContextualPreferenceManager contextualPreferenceManager;
+    private final MessageHelper messageHelper;
+    private final AuthManager authManager;
+
+    public void checkRunLaunchLimits(final Integer newInstancesCount) {
+        if (authManager.isAdmin()) {
+            return;
+        }
+        final PipelineUser user = authManager.getCurrentUser();
+        final String userName = user.getUserName();
+        final List<ContextualPreference> allPreferences = contextualPreferenceManager.loadAll();
```

Comment:
I'd suggest to load only group/user related preferences, even if it will be two requests. We can have thousands of preferences and it seems to be overwhelming to load everything for such check.